enhancements/security/api-tls-curves-config.md

```diff
@@ -0,0 +1,164 @@
+ 2. Improve the adaptability of the cluster's TLS configuration to provide support for the constantly evolving TLS landscape.
+
+### Non-Goals
+
+1. Overhauling the current process of TLS configuration in OpenShift. This change merely extends the current TLS options.
+
+## Proposal
+
+This proposal is to expose the ability to specify the TLS curves used in OpenShift components to the OpenShift administrator.
+Currently, administrators can specify a custom TLS profile where they can specifically set which TLS ciphersuites and the minimum TLS version as opposed to using one of the preconfigured TLS profiles. Specifying the set of supported TLS curves will mirror this process of setting [supported ciphers and the minimum TLS version](https://github.com/openshift/api/blob/138912d4ee9944c989f593c51f15c41908155856/config/v1/types_tlssecurityprofile.go#L206). 
+
+The current state of the OpenShift TLS stack uses a default set of curves with no way to specify them. This eases the burden on administators, however new quantum secure algorithms rely on a set of curves outside of the conventional default curves. For example, curves like [ML-KEM](https://www.ietf.org/archive/id/draft-connolly-tls-mlkem-key-agreement-05.html) provide a quantum safe mechanism for sharing secrets necessary for the TLS handshake, whereas curves like [X22519](https://datatracker.ietf.org/doc/html/rfc7748) (a commonly used conventional curve) are [weak against quantum computing](https://crypto.stackexchange.com/questions/59770/how-effective-is-quantum-computing-against-elliptic-curve-cryptography).
+
+The ability to set curves explicitely will also make it possible to align our 
+OpenShift TLS profiles to match the curves present in the [Mozilla TLS Profiles](https://wiki.mozilla.org/Security/Server_Side_TLS). 
+
+This change will require working with OpenShift component owners to use this new field, however this is outside the scope of this proposal.
+
+### Workflow Description
+
+Administrators will use the [existing custom TLS security profile flow](https://docs.redhat.com/en/documentation/openshift_container_platform/4.20/html/security_and_compliance/tls-security-profiles#tls-profiles-ingress-configuring_tls-security-profiles) for setting the supported curves. 
+
+Specifically administrators will use 
+
+`oc edit IngressController default -n openshift-ingress-operator`
+
+and edit the spec.tlsSecurityProfile field:
+
+```
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+ ...
+spec:
+  tlsSecurityProfile:
+    type: Custom 
+    custom: 
+      ciphers: 
+      - ECDHE-RSA-CHACHA20-POLY1305
+      minTLSVersion: VersionTLS13
+      curves:
+      - X25519MLKEM512
+ ...
+```
+
+### API Extensions
+
+- Adds a `curves` field to the `spec.tlsSecurityProfile` 
+- The addition of this field should not affect existing API behaviour
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+N/A
```

Comment:
Is this supported on HyperShift? Are there any special considerations for HyperShift here?